### PR TITLE
Update logic to match GPUs only for optimized model selection

### PR DESCRIPTION
**Pull request by shivamerla**

### CODE REVIEW

**Comment by mikemckiernan** [comment]
I know you didn't do this for giggles...what happens if someone specifies `engine: vllm`?  Or is the goal to remove that too?  (Not urgent, I'll catch on eventually.)

File: internal/nimparser/nimparser.go [hunk 1]
```diff
@@ -166,6 +177,10 @@ func MatchProfiles(modelSpec appsv1alpha1.ModelSpec, manifest NIMManifest, disco
 	return selectedProfiles, nil
 }
 
+func isOptimizedEngine(engine string) bool {
+	return engine != "" && strings.Contains(strings.ToLower(engine), BackendTypeTensorRT)
```
Comment (slu2011) [inline internal/nimparser/nimparser.go:181]: Are we going to have additional backend types in the future? If yes please add todo....

File: internal/nimparser/nimparser.go [hunk 2]
```diff
@@ -152,11 +157,17 @@ func MatchProfiles(modelSpec appsv1alpha1.ModelSpec, manifest NIMManifest, disco
 			continue
 		}
 
-		// GPU matching logic
-		if len(modelSpec.GPUs) > 0 || len(discoveredGPUs) > 0 {
-			if !matchGPUProfile(modelSpec, profile, discoveredGPUs) {
+		// Perform GPU match only when optimized engine is selected or GPU filters are provided
+		if isOptimizedEngine(modelSpec.Engine) || len(modelSpec.GPUs) > 0 {
```
Comment (slu2011) [inline internal/nimparser/nimparser.go:161]: A model may not use tensorrt engine while still requires gpu, is that right? Seems that for that case there would be no match

Comment (shivamerla) [reply internal/nimparser/nimparser.go:161]: No, this case is to still filter optimized profiles, when GPU options are provided, but not the engine type itself in the model spec.

File: internal/nimparser/nimparser.go (hunk 1 above)
Comment (shivamerla) [reply internal/nimparser/nimparser.go:181]: I am not sure on that, we can add if NVIDIA builds for other backends.

**Comment by shivamerla** [comment]
> I know you didn't do this for giggles...what happens if someone specifies `engine: vllm`? Or is the goal to remove that too? (Not urgent, I'll catch on eventually.)

If it is explicitly specified, we match with only those backends/engines. If not, we assume "tensorrt" when GPU/ID filtering is provided, as that only applies to optimized models.

**Review by slu2011 — APPROVED** [review]

**Pull request merged**